src/Storages/ObjectStorage/StorageObjectStorageConfiguration.cpp

```diff
@@ -72,38 +94,88 @@ void StorageObjectStorageConfiguration::initialize(
     configuration_to_initialize.initialized = true;
 }
 
-void StorageObjectStorageConfiguration::check(ContextPtr) const
+void StorageObjectStorageConfiguration::initPartitionStrategy(ASTPtr partition_by, const ColumnsDescription & columns, ContextPtr context)
 {
-    FormatFactory::instance().checkFormatName(format);
+    partition_strategy = PartitionStrategyFactory::get(
+        partition_strategy_type,
+        partition_by,
+        columns.getOrdinary(),
+        context,
+        format,
+        getRawPath().withGlobs(),
+        getRawPath().withPartitionWildcard(),
+        partition_columns_in_data_file);
+
+    if (partition_strategy)
+    {
+        LOG_DEBUG(getLogger("StorageObjectStorageConfiguration"), "Initialized partition strategy {}", magic_enum::enum_name(partition_strategy_type));
+    }
+}
+
+StorageObjectStorageConfiguration::Path StorageObjectStorageConfiguration::getPathForRead() const
```

Comment:
I am still thinking that may be it makes sense to preinitialize path_for_read as it is not supposed to change at all unlike write path, and so we'll return Path const reference instead of a copy...